### DISCUSSION TITLE
pnpm install duplicate package with same version

### DISCUSSION THREAD

**Discussion by tqma113** [pnpm/pnpm#3897] [Q&A]
As the title, pnpm install duplicate package with same version

```md
@modern-js/plugin-bff -> @modern-js/bff-utils
@modern-js/plugin-express -> @modern-js/bff-utils
```

will get two package `@modern-js/bff-utils` with same version and different hash

![image](https://user-images.githubusercontent.com/22792671/138027073-332447ce-af50-495d-afad-835d5baf048f.png)

Is it a feature? Is there any way to keep it single not duplicate?

**Comment by thynson** [comment]
Their hash postfixes indicate that they're installed with different peer dependencies.

**Reply by zkochan** [reply]
This answer is correct, `@modern-js/bff-utils` resolved the peer dependencies differently in those two places.

**Reply by targeral** [reply]
If I want to use the same @modern-js/bff-utils instance in both packages, what should I do? Two copies of the same version of the content have caused some problems.

**Reply by zkochan** [reply]
probably you should look what are the peer dependencies of `@modern-js/bff-utils` and force some specific versions of them using overrides (https://pnpm.io/package_json#pnpmoverrides) then run `pnpm install`

**Reply by xushimin1207** [reply]
If a package has no peer dependencies, does that mean it doesn't have this problem

**Comment by lanmch** [comment]
hi I have the same problem, do you hav solve it?

**Comment by xiaoxiaojx** [comment]
The cause of the problem and the solution can be seen in this 👇
https://github.com/pnpm/pnpm/discussions/4051
